src/bin/fullnode.rs

```diff
@@ -126,30 +126,48 @@ fn main() {
         }
     }
 
+    let mut local_gossip_addr = bind_addr.clone();
+    local_gossip_addr.set_port(repl_data.gossip_addr.port());
```

Comment:
So ugly!  You thinking ReplicatedData should separate IP and ports into separate fields?

Reply:
🙈 yeah something like that would totally help.  That was more Rust than I'm comfortable busting out at this point, you're ok with this as is for now?